# Case Study on Measures of Central Tendency and Dispersion

In [1]:
#importing relevant libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

In [21]:
#reading and displaying the dataset
data = pd.read_csv('StudentsPerformance.csv')
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [3]:
#dataset info
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


# 1. Find out how many males and females participated in the test 

In [4]:
#count of gender
data.gender.value_counts()

female    518
male      482
Name: gender, dtype: int64

# 2. What do you think about the students' parental level of education?

In [5]:
# count of students parental level of education
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

In [20]:
#impact of scores based on parental level of education
parent=data.groupby('parental level of education')[['math score','reading score','writing score']].mean()
parent

math score  reading score  writing score
parental level of education                                          
associate's degree            67.977477      71.018018      70.031532
bachelor's degree             69.288136      73.000000      73.381356
high school                   61.821429      64.602041      62.403061
master's degree               70.254237      75.949153      75.677966
some college                  67.128319      69.566372      69.035398
some high school              63.134078      66.759777      64.888268

 The students whose parents have masters degree have highest average in all test scores

# 3.  a) Who scores the most on average for math, reading and writing based on gender

In [18]:
avg1=data.groupby('gender')[['math score','reading score','writing score']].mean()
avg1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Summary:For maths, male students scored higher on average, while in reading and writing, female students scored higher on average.

#  b) Who scores the most on average for math, reading and writing based on test preparation course

In [19]:
avgb1=data.groupby('test preparation course')[['math score','reading score','writing score']].mean()
avgb1

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

Summary:Students who completed test preparation course scored higher on average for all three tests. 

# 4. a) What do you think about the scoring variation for math, reading and writing based on gender

In [12]:
var=data.groupby('gender')['math score','reading score','writing score'].var()
var

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

Variation of scores is higher in Female in all 3 scores than male

# b) What do you think about the scoring variation for math, reading and writing based on test preparation course

In [13]:
var=data.groupby('test preparation course')['math score','reading score','writing score'].var()
var

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

Variation of score is higher for who have not completed test preparation course in all scores

# 5. Top 25% based on math score

In [14]:
math = data['math score'].nlargest(int(len(data)*0.25))

In [15]:
pd.DataFrame(data=math)

math score
515         100
516         100
517         100
996         100
997         100
..          ...
428          77
429          77
849          77
850          77
851          77

[250 rows x 1 columns]

Dataframe math gives the 250 students with the highest scores in math test. The dataframe can be used to give bonus points to the respective students.